modules/apps/layout/layout-page-template-api/src/main/resources/com/liferay/layout/page/template/validator/dependencies/page_definition_json_schema.json

```diff
@@ -3,10 +3,68 @@
 	"$schema": "http://json-schema.org/draft-07/schema#",
 	"additionalProperties": false,
 	"definitions": {
+		"ClassNameReference": {
+			"additionalProperties": false,
+			"properties": {
+				"className": {
+					"title": "The className Schema",
+					"type": "string"
+				}
+			},
+			"required": [
+				"className"
+			],
+			"title": "The ClassNameReference Schema",
+			"type": "object"
+		},
+		"ClassPKItemReference": {
+			"additionalProperties": false,
+			"properties": {
+				"itemClassName": {
+					"title": "The itemClassName Schema",
+					"type": "string"
+				},
+				"itemClassPK": {
+					"title": "The itemClassPK Schema",
+					"type": "integer"
+				}
+			},
+			"required": [
+				"itemClassName",
+				"itemClassPK"
+			],
+			"title": "The ClassPKItemReference Schema",
+			"type": "object"
+		},
 		"CollectionDefinition": {
 			"additionalProperties": false,
 			"properties": {
 				"collectionConfig": {
+					"properties": {
+						"collectionReference": {
+							"oneOf": [
+								{
+									"$ref": "#/definitions/ClassNameReference"
+								},
+								{
+									"$ref": "#/definitions/ClassPKItemReference"
+								}
+							],
+							"title": "The collectionReference Schema",
+							"type": "object"
+						},
+						"collectionType" : {
```

Comment:
This might be a small thing that we have to be ok with it, but it's not ideal that the type of reference above is really tied to the value of collectionType, but the schema does not enforce that in any way. 
Can you think of a way to solve that?

Reply:
That's a very good point. We can tie the value of collectionType to the type of the reference applying subschemas conditionally (wrapping pairs of `if` and `then` inside an `allOf`). 
That would be consistent with what we are already doing for `type` and `definition` under `PageElement`.

Reply:
I don't fully understand how it would work (it's easier for me with an example), but if we are already doing it elsewhere it seems it's the way to go :)

Reply:
Here is a link to the relevant part of the JSON schema [documentation](https://json-schema.org/understanding-json-schema/reference/conditionals.html).

And this is how it will look like for collectionConfig:

"collectionConfig": {
	"additionalProperties": false,
	"allOf": [
		{
			"if": {
				"properties": {
					"collectionType": {
						"const": "Collection"
					}
				}
			},
			"then": {
				"properties": {
					"collectionReference": {
						"$ref": "#/definitions/ClassPKReference"
					}
				}
			}
		},
		{
			"if": {
				"properties": {
					"collectionType": {
						"const": "CollectionProvider"
					}
				}
			},
			"then": {
				"properties": {
					"collectionReference": {
						"$ref": "#/definitions/ClassNameReference"
					}
				}
			}
		}
	],
	"properties": {
		"collectionReference": {
			"oneOf": [
				{
					"$ref": "#/definitions/ClassNameReference"
				},
				{
					"$ref": "#/definitions/ClassPKReference"
				}
			],
			"title": "The collectionReference Schema",
			"type": "object"
		},
		"collectionType": {
			"enum": [
				"Collection",
				"CollectionProvider"
			],
			"title": "The collectionType Schema",
			"type": "object"
		}
	},
	"required": [
		"collectionReference",
		"collectionType"
	],
	"title": "The collectionConfig Schema",
	"type": "object"
}

Reply:
Got it.
If you think that's the best solution for the need, then go ahead with it.